### PR TITLE
Object-Oriented JavaScript (Frogger)

### PR DESCRIPTION
**Pull request by akulahere**
### Object-Oriented JavaScript

[Demo](https://akulahere.github.io/frontend-nanodegree-arcade-game/)
[Code base](https://github.com/akulahere/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@akulahere commit 2733fae will remove your other task code from the repo if this PR gets merged after your other task PR.
We do not want this.
We need to fix this situation.
Please feel free asking for help in students' chat.

**Comment by akulahere** [comment]
@OleksiyRudenko I reverted unnecessary commits. I hope it looks fine now.

**Comment by OleksiyRudenko** [comment]
The file of interest still gets deleted
image

**Comment by akulahere** [comment]
I am not sure that I chose the right way (force push), but I don't understand how else I could delete commits in the remote branch if git revert didn't help

**Comment by akulahere** [comment]
What I have done:
- rebase localy
- git push origin -f

**Comment by OleksiyRudenko** [comment]
This worked just fine 👍 

When creating a branch for your next task make sure to [follow the instructions](https://github.com/kottans/frontend-2021-homeworks#b1-submitting-new-task-app-code-base) to avoid similar situations.

File: submissions/akulahere/frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+let Player = function (x,y) {
+  this.x = x;
+  this.y = y;
+  this.sprite = 'images/char-boy.png';
+}
+
+Player.prototype.update = function () {
+  allEnemies.forEach(enemy => {
```
Comment (OleksiyRudenko) [inline submissions/akulahere/frogger-game/app.js:46]: Classes should not refer to any global variables, like global variable `player`, which is an instance of `Player` class, or `allEnemies`. Handle those just like any other values from the world outside `Player` or `Enemy`.

Referring to global constants and globals provided by the gaming platform like `Resources` is OK.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@akulahere well done.
There is an issue to fix though.

File: submissions/akulahere/frogger-game/app.js (hunk 1 above)
Comment (OleksiyRudenko) [reply submissions/akulahere/frogger-game/app.js:46]: Also this method does two different things:
1. Updates a player's position (as method name promises)
2. Checks for collisions (method name doesn't promise this)

This breaks Single Responsibility Principle. Please resolve.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@akulahere yet another important thing -- SRP.

File: submissions/akulahere/frogger-game/app.js (hunk 1 above)
Comment (akulahere) [reply submissions/akulahere/frogger-game/app.js]: Done.
isCollision function added and update method refactored.

**Review by OleksiyRudenko — APPROVED** [review]
@akulahere great job!

**Pull request merged**